packages/plugins/src/plugins/morpho/vaultsJob.ts

```diff
@@ -0,0 +1,47 @@
+import { EvmNetworkIdType } from '@sonarwatch/portfolio-core';
+import { getVaults } from './utils/api';
+import { Job, JobExecutor } from '../../Job';
+import { morphoVaultsCachePrefix, platformId } from './constants';
+import { Cache } from '../../Cache';
+import { tokenListInfoPrefix } from '../tokens/constants';
+import { buildTokenMetaDataItems, buildTokenPriceSources } from './helpers';
+import { SEVEN_DAYS } from '../../utils/octav/time';
+
+export function vaultsJob(networkId: EvmNetworkIdType): Job {
+  const executor: JobExecutor = async (cache: Cache) => {
+    const vaults = await getVaults(networkId);
+    if (vaults.length === 0) {
```

Comment:
Should we be extra careful ?
```suggestion
    if (vaults || vaults.length === 0) {
```

Reply:
To be similar to the below fetcher ?

Reply:
No it will always return an array